File: tests/fixtures/vulnerable/js/weak-auth.js
// Vulnerable Authentication Implementation
// Should trigger: weak-password-validation, missing-rate-limiting, insecure-session-config, missing-jwt-verification

const express = require('express');
const jwt = require('jsonwebtoken');
const app = express();

// Issue 1: Weak password validation
function isValidPassword(password) {
  // Only checking if length is greater than 5 - too weak!
  if (password.length < 6) {
    return false;
  }
  return true;
}

// Issue 2: Missing rate limiting on login endpoint
app.post('/login', async (req, res) => {
  const { username, password } = req.body;
  
  if (!isValidPassword(password)) {
    return res.status(400).json({ error: 'Invalid password' });
  }
  
  const user = await db.findUser(username, password);
  
  if (user) {
    // Issue 3: Insecure session cookie - missing httpOnly, secure, sameSite
    res.cookie('sessionId', user.sessionToken);
    
    // Issue 4: JWT decoded without verification
    const tokenData = jwt.decode(user.token);
    
    res.json({ success: true, user: tokenData });
  } else {
    res.status(401).json({ error: 'Authentication failed' });
  }
});

// Another weak password check
function validateUserPassword(pwd) {
  return pwd.length >= 4; // Even weaker!
}

module.exports = { isValidPassword, validateUserPassword };

File: tests/fixtures/vulnerable/py/weak-auth.py
# Vulnerable Authentication Implementation (Python)
# Should trigger: weak-password-validation, missing-rate-limiting, insecure-session-config, missing-jwt-verification

from flask import Flask, request, jsonify, make_response
import jwt

app = Flask(__name__)

# Issue 1: Weak password validation
def is_valid_password(password):
    # Only checking length - too weak!
    if len(password) < 6:
        return False
    return True

def check_password_strength(pwd):
    # Even weaker - only 4 characters
    return len(pwd) >= 4

# Issue 2: Missing rate limiting on login endpoint
@app.route('/login', methods=['POST'])
def login():
    username = request.json.get('username')
    password = request.json.get('password')
    
    if not is_valid_password(password):
        return jsonify({'error': 'Invalid password'}), 400
    
    user = db.find_user(username, password)
    
    if user:
        response = make_response(jsonify({'success': True}))
        
        # Issue 3: Insecure session configuration
        response.set_cookie('sessionId', user.session_token)
        # Missing: httponly=True, secure=True, samesite='Strict'
        
        # Issue 4: JWT decoded without verification
        token_data = jwt.decode(user.token, verify=False)
        
        return response
    
    return jsonify({'error': 'Authentication failed'}), 401

# Issue 3: Insecure Flask session config
app.config['SESSION_COOKIE_SECURE'] = False
app.config['SESSION_COOKIE_HTTPONLY'] = False
app.config['SESSION_COOKIE_SAMESITE'] = None

if __name__ == '__main__':
    app.run()
